### PR TITLE
[swift5] Remove optional from body

### PR DESCRIPTION
**Pull request by funzin**
Remove optional from `body`
When using `getNonDecodableBuilder`, Response body type is `Void` only so we don't need to have `body` as optional.
- Remove unused switch case code
- Change from `body: T?` to `body: T`

we won't have to do any unnecessary unwrapping when using api response.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by funzin** [comment]
@4brunu 
Please review it when you have time.

File: modules/openapi-generator/src/main/resources/swift5/libraries/alamofire/AlamofireImplementations.mustache [hunk 1]
```diff
@@ -160,100 +160,20 @@ private var managerStore = SynchronizedDictionary<String, Alamofire.Session>()
         let validatedRequest = request.validate()
 
         switch T.self {
-        case is String.Type:
```
Comment (funzin) [inline modules/openapi-generator/src/main/resources/swift5/libraries/alamofire/AlamofireImplementations.mustache:163]: When response type is void, getNonDecodableBuilder is used
- https://github.com/OpenAPITools/openapi-generator/blob/126e4061937fabd27a1ac020ca4dcfadb7b1e357/modules/openapi-generator/src/main/resources/swift5/api.mustache#L454
- https://github.com/OpenAPITools/openapi-generator/blob/126e4061937fabd27a1ac020ca4dcfadb7b1e357/modules/openapi-generator/src/main/resources/swift5/api.mustache#L163

**Comment by 4brunu** [comment]
Hi @funzin, thanks for creating this PR.
CI is failing because there are places where the optional body was being unwrapped, but now since it's not optional anymore, it can't be unwrapped.
Could you please check for the other places that need to be updated?
For example the file `api.mustache` and i'm not sure if there are other places.
Thanks

**Comment by funzin** [comment]
OK, I will check these places:+1:

File: modules/openapi-generator/src/main/resources/swift5/api.mustache [hunk 2]
```diff
@@ -167,7 +167,7 @@ extension {{projectName}}API {
                 switch result {
                 {{#returnType}}
                 case let .success(response):
-                    promise(.success(response.body!))
+                    promise(.success(response.body))
```
Comment (4brunu) [inline modules/openapi-generator/src/main/resources/swift5/api.mustache:170]: @funzin please search and replace all occurrences of `response.body!` with `response.body` in this class, to fix all the cases. Thanks

Comment (funzin) [reply modules/openapi-generator/src/main/resources/swift5/api.mustache:170]: I have already checked `response.body!` in `swift5`
dbdc585

Maybe, CI will be succeed.

Comment (funzin) [reply modules/openapi-generator/src/main/resources/swift5/api.mustache:170]: @4brunu 
CI ✅

File: modules/openapi-generator/src/main/resources/swift5/libraries/alamofire/AlamofireImplementations.mustache [hunk 3]
```diff
@@ -160,100 +160,20 @@ private var managerStore = SynchronizedDictionary<String, Alamofire.Session>()
...
-
-                switch dataResponse.result {
-                case .success:
-                    completion(.success(Response(response: dataResponse.response!, body: dataResponse.data as? T)))
-                case let .failure(error):
-                    completion(.failure(ErrorResponse.error(dataResponse.response?.statusCode ?? 500, dataResponse.data, dataResponse.response, error)))
-                }
-
-            })
+            fatalError("Unsupported Resopnse Body Type - \(String(describing: T.self))")
```
Comment (4brunu) [inline modules/openapi-generator/src/main/resources/swift5/libraries/alamofire/AlamofireImplementations.mustache:176]: @funzin typo on "Resopnse"

File: modules/openapi-generator/src/main/resources/swift5/libraries/urlsession/URLSessionImplementations.mustache [hunk 4]
```diff
@@ -200,60 +200,12 @@ private var credentialStore = SynchronizedDictionary<Int, URLCredential>()
...
-
-        case is Data.Type:
-
-            completion(.success(Response(response: httpResponse, body: data as? T)))
+            completion(.success(Response(response: httpResponse, body: () as! T)))
 
         default:
-
-            completion(.success(Response(response: httpResponse, body: data as? T)))
+            fatalError("Unsupported Resopnse Body Type - \(String(describing: T.self))")
```
Comment (4brunu) [inline modules/openapi-generator/src/main/resources/swift5/libraries/urlsession/URLSessionImplementations.mustache:208]: @funzin typo on "Resopnse"

File: modules/openapi-generator/src/main/resources/swift5/libraries/alamofire/AlamofireImplementations.mustache (hunk 3 above)
Comment (funzin) [reply modules/openapi-generator/src/main/resources/swift5/libraries/alamofire/AlamofireImplementations.mustache]: done
c35858c

File: modules/openapi-generator/src/main/resources/swift5/libraries/urlsession/URLSessionImplementations.mustache (hunk 4 above)
Comment (funzin) [reply modules/openapi-generator/src/main/resources/swift5/libraries/urlsession/URLSessionImplementations.mustache]: done
c35858c

**Review by 4brunu — APPROVED** [review]
Looks good to me 👍 
Thanks for this PR

**Pull request merged**